src/main/java/by/custom/utilcalculator/service/UserProgressManager.java

```diff
@@ -33,93 +35,123 @@ public void createNewUserProgress(final String chatID) throws UtilsborException
+            userProgress.setCarAge(CarAge.LESS_OR_3_YEARS);
+        } else {
+            if (Objects.equals(requestingCommand, Command.MORE_THAN_3_YEARS_AGE.getCommand())) {
+                userProgress.setCarAge(CarAge.MORE_3_YEARS);
+            }
         }
+
         userProgressStorage.save(userProgress);
         message = messagesCreator.getSummaryAnswer(userProgress);
         return message;
     }
 
     public String processEngineType(final String requestingCommand, final String chatID) throws UtilsborException {
-        UserProgress userProgress;
+        final UserProgress userProgress;
         userProgress = userProgressStorage.get(chatID);
 
         if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
             throw new InvalidOrderCommandException(chatID, requestingCommand);
         }
 
-        String message;
-        switch (requestingCommand) {
-            case Command.GASOLINE_TYPE_ENGINE -> userProgress.setTypeOfEngine(TypeOfEngine.GASOLINE);
-            case Command.ELECTRIC_TYPE_ENGINE -> userProgress.setTypeOfEngine(TypeOfEngine.ELECTRIC);
+        if (Objects.equals(requestingCommand, Command.GASOLINE_TYPE_ENGINE.getCommand())) {
+            userProgress.setTypeOfEngine(TypeOfEngine.GASOLINE);
+        } else {
+            if (Objects.equals(requestingCommand, Command.ELECTRIC_TYPE_ENGINE.getCommand())) {
+                userProgress.setTypeOfEngine(TypeOfEngine.ELECTRIC);
+            }
         }
+        final String message;
         userProgressStorage.save(userProgress);
         message = messagesCreator.getSummaryAnswer(userProgress);
         return message;
     }
 
     public String processEngineVolume(final String requestingCommand, final String chatID) throws UtilsborException {
-        UserProgress userProgress;
+        final UserProgress userProgress;
         userProgress = userProgressStorage.get(chatID);
 
-        if (!UserProgressValidator.validateCommand(Command.VOLUME, userProgress)) {
+        if (!UserProgressValidator.validateCommand(Command.VOLUME.getCommand(), userProgress)) {
             throw new InvalidOrderCommandException(chatID, requestingCommand);
         }
 
-        String message;
-        switch (requestingCommand) {
-            case Command.VOLUME_LESS_1000_CM -> userProgress.setVolumeOfEngine(VolumeOfEngine.LESS_1000);
-            case Command.VOLUME_BETWEEN_1000_2000_CM ->
-                    userProgress.setVolumeOfEngine(VolumeOfEngine.BETWEEN_1000_AND_2000);
-            case Command.VOLUME_BETWEEN_2000_3000_CM ->
-                    userProgress.setVolumeOfEngine(VolumeOfEngine.BETWEEN_2000_AND_3000);
-            case Command.VOLUME_BETWEEN_3000_3500_CM ->
-                    userProgress.setVolumeOfEngine(VolumeOfEngine.BETWEEN_3000_AND_3500);
-            case Command.VOLUME_MORE_3500_CM -> userProgress.setVolumeOfEngine(VolumeOfEngine.MORE_3500);
+        final String message;
+
+        while (true) {
```

Comment:
switch case is better. And why loop is needed is not clear.

Reply:
It was impossible to use switch-case because I used Strings - it is forbidden to use String in switch-case. 
Using string here was a mistake, I replaced it by switch-case.

Infinity loop was needed here because it made code more beautiful - I hadn't use long 
if(){
}else{
if() {
}else {
if()}}